feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle/configurator-attribute-single-selection-bundle.component.ts

```diff
@@ -1,128 +1,15 @@
   templateUrl:
     './configurator-attribute-single-selection-bundle.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConfiguratorAttributeSingleSelectionBundleComponent extends ConfiguratorAttributeBaseComponent {
-  loading$ = new BehaviorSubject<boolean>(false);
-
-  @Input() attribute: Configurator.Attribute;
-  @Input() ownerKey: string;
-  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
-
-  constructor(protected quantityService: ConfiguratorAttributeQuantityService) {
-    super();
-  }
-
-  /**
-   * Checks if we are supposed to render a quantity control, which
-   * can be derived from the attribute meta data
-   *
-   * @return {boolean} - Display quantity picker?
-   */
-  get withQuantity(): boolean {
-    return this.quantityService.withQuantity(
-      this.attribute.dataType,
-      this.attribute.uiType
-    );
-  }
-
-  /**
-   * Checks if quantity control should be disabled
-   *
-   * @return {boolean} - Disable quantity picker?
-   */
-  get disableQuantityActions(): boolean {
-    return this.quantityService.disableQuantityActions(
-      this.attribute.selectedSingleValue
-    );
-  }
-
-  onSelect(value: string): void {
-    this.loading$.next(true);
-
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: value,
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-
-    this.selectionChange.emit(event);
-  }
-
-  onDeselect(): void {
-    this.loading$.next(true);
-
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: '',
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-
-    this.selectionChange.emit(event);
-  }
-
-  onHandleQuantity(quantity: any): void {
-    this.loading$.next(true);
-
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        quantity,
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE_QUANTITY,
-    };
-
-    this.selectionChange.emit(event);
-  }
-
-  onChangeQuantity(eventObject: any): void {
-    if (!eventObject.quantity) {
-      this.onDeselect();
-    } else {
-      this.onHandleQuantity(eventObject.quantity);
-    }
-  }
-
-  /**
-   * Extract corresponding price formula parameters
-   *
-   * @return {ConfiguratorPriceComponentOptions} - New price formula
-   */
-  extractPriceFormulaParameters(): ConfiguratorPriceComponentOptions {
-    return {
-      quantity: this.attribute.quantity,
-      price: this.getSelectedValuePrice(),
-      priceTotal: this.attribute.attributePriceTotal,
-      isLightedUp: true,
-    };
-  }
-
+export class ConfiguratorAttributeSingleSelectionBundleComponent extends ConfiguratorAttributeSingleSelectionBaseComponent {
```

Comment:
can `ConfiguratorAttributeSingleSelectionBundleDropdownComponent` also extend this component?

Reply:
I will check it